shell/platform/darwin/ios/framework/Source/accessibility_bridge.mm

```diff
@@ -220,57 +197,513 @@ - (CGRect)accessibilityFrame {
   // the logical pixel coordinate system. Therefore, we divide by the `scale` (pixel ratio) to
   // convert.
-  CGFloat scale = [[_bridge->view() window] screen].scale;
+  CGFloat scale = [[[self bridge]->view() window] screen].scale;
   auto result =
       CGRectMake(rect.x() / scale, rect.y() / scale, rect.width() / scale, rect.height() / scale);
-  return UIAccessibilityConvertFrameToScreenCoordinates(result, _bridge->view());
+  return UIAccessibilityConvertFrameToScreenCoordinates(result, [self bridge]->view());
 }
 
 #pragma mark - UIAccessibilityElement protocol
 
 - (id)accessibilityContainer {
-  if ([self hasChildren] || _uid == kRootNodeId) {
+  if ([self hasChildren] || [self uid] == kRootNodeId) {
     if (_container == nil)
-      _container = [[SemanticsObjectContainer alloc] initWithSemanticsObject:self bridge:_bridge];
+      _container =
+          [[SemanticsObjectContainer alloc] initWithSemanticsObject:self bridge:[self bridge]];
     return _container;
   }
-  NSAssert(_parent != nil, @"Illegal access to non-existent parent of root semantics node");
-  return [_parent accessibilityContainer];
+  NSAssert([self parent] != nil, @"Illegal access to non-existent parent of root semantics node");
+  return [[self parent] accessibilityContainer];
 }
 
 #pragma mark - UIAccessibilityAction overrides
 
 - (BOOL)accessibilityActivate {
-  if (!_node.HasAction(blink::SemanticsAction::kTap))
+  if (![self node].HasAction(blink::SemanticsAction::kTap))
     return NO;
-  _bridge->DispatchSemanticsAction(_uid, blink::SemanticsAction::kTap);
+  [self bridge]->DispatchSemanticsAction([self uid], blink::SemanticsAction::kTap);
   return YES;
 }
 
 - (void)accessibilityIncrement {
-  if (_node.HasAction(blink::SemanticsAction::kIncrease)) {
-    _node.value = _node.increasedValue;
-    _bridge->DispatchSemanticsAction(_uid, blink::SemanticsAction::kIncrease);
+  if ([self node].HasAction(blink::SemanticsAction::kIncrease)) {
+    [self node].value = [self node].increasedValue;
+    [self bridge]->DispatchSemanticsAction([self uid], blink::SemanticsAction::kIncrease);
   }
 }
 
 - (void)accessibilityDecrement {
-  if (_node.HasAction(blink::SemanticsAction::kDecrease)) {
-    _node.value = _node.decreasedValue;
-    _bridge->DispatchSemanticsAction(_uid, blink::SemanticsAction::kDecrease);
+  if ([self node].HasAction(blink::SemanticsAction::kDecrease)) {
+    [self node].value = [self node].decreasedValue;
+    [self bridge]->DispatchSemanticsAction([self uid], blink::SemanticsAction::kDecrease);
   }
 }
 
 - (BOOL)accessibilityScroll:(UIAccessibilityScrollDirection)direction {
   blink::SemanticsAction action = GetSemanticsActionForScrollDirection(direction);
-  if (!_node.HasAction(action))
+  if (![self node].HasAction(action))
     return NO;
-  _bridge->DispatchSemanticsAction(_uid, action);
+  [self bridge]->DispatchSemanticsAction([self uid], action);
   return YES;
 }
 
 @end
 
+@interface InactiveTextPosition : UITextPosition
```

Comment:
Why can we not reuse the TextPosition and TextRange from the active text input?

Reply:
Is there every any danger that iOS asks us for a TextPosition or TextRange while the text field is unfocused (returning an InactiveTextPosition/Range) and then when the text field becomes focused iOS passes those InactiveTextPositions/Ranges back to the real text input view, which doesn't know what to do with them and potentially crashes?

Reply:
Done.